### DISCUSSION TITLE
How to run solara in a docker container

### DISCUSSION THREAD

**Discussion by saurabhmangal** [widgetti/solara#220] [Q&A]
I am not able to run solara using a docker container. I get the following error
ERROR:    [Errno 99] error while attempting to bind on address ('::1', 8765, 0, 0): cannot assign requested address

Can someone help me with this

**Comment by maartenbreddels** [comment]
Hi,

See https://github.com/widgetti/solara/issues/112 . Let me know if that answers your question.

Regards,

Maarten

**Reply by maartenbreddels** [reply]
And now in the docs https://solara.dev/docs/deploying/self-hosted

**Reply by rh-afk** [reply]
hi @maartenbreddels how to run solara server using self hosted server since when I run the server , its just run in localhost and I couldn't open the apps from another machine in the network. can we have similarity with panel or bokeh for example run with this in command line jupyter lab server:
panel serve app_script.py --allow-websocket-origin foo.com:8081

then I can just type in my computer the address of foo.com:8081. 
current practice of solara server run only in [localhost.](https://github.com/widgetti/solara/discussions/248)

kindly please advise

**Reply by rh-afk** [reply]
Hi all,

I just read solara help and we can specify host and port for run solara server. Perfect.
![image](https://github.com/widgetti/solara/assets/141318843/3af3987e-34aa-4640-8000-60dc62978479)
